### PR TITLE
35: work on shared file management

### PR DESCRIPTION
**Pull request by Mathieu414**
Migrated the shared files page to the ORM.
Rewrote pretty much everything in the Uploadfield class of the validator. Now using the "error" field in the `$_FILES` to validate the upload. Much better practice !
We need to discuss the file name validation, not sure if it is useful.

### CODE REVIEW

File: database/models/files.db.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public DateTime $date;
+
+    #[Column]
+    public string $mime = "";
+
+    #[Column]
+    public bool $is_public = true;
+
+    #[ManyToOne]
+    public Race|null $race = null;
```
Comment (Atmos4) [inline database/models/files.db.php:28]: Keep in mind you need to add the other side of the relation to `Event` and `Race` entities so that you can just join them easily. You can do this in another PR, try to take inspiration from the relation between `Event` and `Race`

File: bin/doctrine [hunk 2]
```diff
@@ -36,5 +36,6 @@ $commands = [
     new Command\GenerateCommand($deps),
     new Command\MigrateCommand($deps),
     new Command\DiffCommand($deps),
+    new Command\ListCommand($deps),
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply bin/doctrine:39]: This can be done in another PR :)

File: components/form/form_validation.php [hunk 3]
```diff
@@ -66,7 +66,7 @@ function render_validation()
         }
         if (!$this->empty) {
             if ($this->valid() && $this->success) {
-                $result .= "<ins>$this->success</ins>";
+                $result .= "<ins>$this->success</ins><br></br>";
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply components/form/form_validation.php:69]: Didn't notice but slight problem: `br` is self closing
```suggestion
                $result .= "<ins>$this->success</ins><br/>";
```

File: components/form/form_validation.php [hunk 4]
```diff
@@ -447,78 +447,98 @@ function set_type()
...
-                $this->check_format($fileType);
-            } else {
-                $this->set_error("Choisissez un fichier");
-            }
-
-        }
-    }
+                $this->file_name = $_FILES[$this->key]["name"] ?? "";
+                $this->target_file = $this->target_dir . basename($_FILES[$this->key]["name"]);
+                $this->file_type = strtolower(pathinfo($this->target_file, PATHINFO_EXTENSION));
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply components/form/form_validation.php:461]: A good solution would be to move all variable initializations in a constructor

[the remark this answers is not in the document]
Comment (Mathieu414) [reply components/form/form_validation.php:461]: > A good solution would be to move all variable initializations in a constructor

It's exactly what I did, thanks for the advice !

File: components/form/form_validation.php [hunk 5]
```diff
@@ -441,84 +441,112 @@ function render(string $attrs = "")
...
-            $this->set_error("Fichier trop lourd.");
-        }
-    }
+        var_dump($_FILES);
+        if (isset($_FILES[$this->key])) {
+            if ($_FILES[$this->key]["name"] != '') {
+                // Check if the error field is ok 
+                if (
+                    !isset($_FILES[$this->key]['error']) ||
+                    is_array($_FILES[$this->key]['error'])
```
Comment (Atmos4) [inline components/form/form_validation.php:484]: This is wrong. It should be `!is_array`

Comment (Atmos4) [reply components/form/form_validation.php:484]: You probably want to refactor this part in fact.

Comment (Atmos4) [reply components/form/form_validation.php:484]: Nevermind I didn't realize this should NOT be an array :) LGTM

**Review by Atmos4 — APPROVED** [review]
